src/contexts/socket.context.tsx

```diff
@@ -79,6 +81,13 @@ function SocketProvider({ children }: SocketProviderProps) {
         );
       });
 
+      state.socket.on("build_deleted", function (build: Build) {
+        clearBuild(buildDispatch, build.id);
+        if (selectedBuild && selectedBuild.id === build.id) {
```

Comment:
could you try reusing logic from `BuildList` not to duplicate it here?
```js
if (indexOfBuildDeleted === indexOfSelectedBuild) {
                  if (buildList.length > 1) {
                    if (indexOfBuildDeleted === 0) {
                      selectBuild(buildDispatch, buildList[1].id);
                    } else {
                      selectBuild(
                        buildDispatch,
                        buildList[indexOfBuildDeleted - 1].id
                      );
                    }
                  } else {
                    selectBuild(buildDispatch, null);
                  }
                }
```

Reply:
Moved the select build after deleting logic to the context now.